pkg/cache/cache.go

```diff
@@ -145,17 +145,19 @@ func (c *Cache) OpenRepository(ctx context.Context, repositorySpec *configapi.Re
 			} else {
 				mbs = git.ErrorIfMissing
 			}
-			if r, err := git.OpenRepository(ctx, repositorySpec.Name, repositorySpec.Namespace, gitSpec, repositorySpec.Spec.Deployment, filepath.Join(c.cacheDir, "git"), git.GitRepositoryOptions{
+
+			r, err := git.OpenRepository(ctx, repositorySpec.Name, repositorySpec.Namespace, gitSpec, repositorySpec.Spec.Deployment, filepath.Join(c.cacheDir, "git"), git.GitRepositoryOptions{
 				CredentialResolver: c.credentialResolver,
 				UserInfoProvider:   c.userInfoProvider,
 				MainBranchStrategy: mbs,
 				UseGitCaBundle:     c.useGitCaBundle,
-			}); err != nil {
+			})
+			if err != nil {
 				return nil, err
-			} else {
-				cachedRepo = newRepository(key, repositorySpec, r, c.objectNotifier, c.metadataStore, c.repoSyncFrequency)
-				c.repositories[key] = cachedRepo
 			}
+
+			cachedRepo = newRepository(key, repositorySpec, r, c.objectNotifier, c.metadataStore, c.repoSyncFrequency)
+			c.repositories[key] = cachedRepo
```

Comment:
previously in the code it seems these 2 lines were making note of this cached repo only when there was no error where as now it is being done regardless of error occurrence.
Is this the intent here? or am i reading this wrong?

Reply:
Because of the if block on lines 155-157, we can never get to line 159 and 160 in an error situation so checking for the error is not required.